commit 9b9afcf7c2a9dc0c2144540221126a61c3787a0f
Author: Michael Still <mikal@stillhq.com>

    Add support for microversion 2.97.

    This microversion adds support for the new SPICE direct console
    type. Note that this patch does not add support for the
    os-console-auth-tokens as python-novaclient is deprecated.

diff --git a/novaclient/__init__.py b/novaclient/__init__.py
index bf021c6a..6629d744 100644
--- a/novaclient/__init__.py
+++ b/novaclient/__init__.py
@@ -25,4 +25,4 @@ API_MIN_VERSION = api_versions.APIVersion("2.1")
 # when client supported the max version, and bumped sequentially, otherwise
 # the client may break due to server side new version may include some
 # backward incompatible change.
-API_MAX_VERSION = api_versions.APIVersion("2.96")
+API_MAX_VERSION = api_versions.APIVersion("2.97")
diff --git a/novaclient/tests/unit/v2/test_shell.py b/novaclient/tests/unit/v2/test_shell.py
index 3f7bc396..25c05053 100644
--- a/novaclient/tests/unit/v2/test_shell.py
+++ b/novaclient/tests/unit/v2/test_shell.py
@@ -4703,6 +4703,7 @@ class ShellTest(utils.TestCase):
             94,  # There are no version-wrapped shell method changes for this.
             95,  # There are no version-wrapped shell method changes for this.
             96,  # There are no version-wrapped shell method changes for this.
+            97,  # Add new spice-direct console type.
         ])
         versions_supported = set(range(0,
                                  novaclient.API_MAX_VERSION.ver_minor + 1))
diff --git a/novaclient/v2/servers.py b/novaclient/v2/servers.py
index 37f55316..b1992e13 100644
--- a/novaclient/v2/servers.py
+++ b/novaclient/v2/servers.py
@@ -37,6 +37,7 @@ CONSOLE_TYPE_ACTION_MAPPING = {
     'novnc': 'os-getVNCConsole',
     'xvpvnc': 'os-getVNCConsole',
     'spice-html5': 'os-getSPICEConsole',
+    'spice-direct': 'os-getSPICEConsole',
     'rdp-html5': 'os-getRDPConsole',
     'serial': 'os-getSerialConsole'
 }
@@ -45,6 +46,7 @@ CONSOLE_TYPE_PROTOCOL_MAPPING = {
     'novnc': 'vnc',
     'xvpvnc': 'vnc',
     'spice-html5': 'spice',
+    'spice-direct': 'spice',
     'rdp-html5': 'rdp',
     'serial': 'serial',
     'webmks': 'mks'
@@ -1128,7 +1130,7 @@ class ServerManager(base.BootingManagerWithFind):

         :param server: server to get console url for
         :param console_type: type can be novnc, xvpvnc, spice-html5,
-                             rdp-html5 and serial.
+                             spice-direct, rdp-html5 and serial.
         """

         action = CONSOLE_TYPE_ACTION_MAPPING.get(console_type)
diff --git a/novaclient/v2/shell.py b/novaclient/v2/shell.py
index 0ca9e794..6fd07305 100644
--- a/novaclient/v2/shell.py
+++ b/novaclient/v2/shell.py
@@ -2885,7 +2885,7 @@ def do_get_vnc_console(cs, args):
 @utils.arg(
     'console_type',
     metavar='<console-type>',
-    help=_('Type of spice console ("spice-html5").'))
+    help=_('Type of spice console ("spice-html5", "spice-direct").'))
 def do_get_spice_console(cs, args):
     """Get a spice console to a server."""
     server = _find_server(cs, args.server)
